src/lc2kFormat.ts

```diff
@@ -1,13 +1,85 @@
 import * as vscode from 'vscode';
+import {TextDecoder} from 'util';
 
 export class Lc2kFormatter implements vscode.DocumentRangeFormattingEditProvider, vscode.DocumentFormattingEditProvider {
   provideDocumentFormattingEdits(document: vscode.TextDocument, options: vscode.FormattingOptions,
                                  token: vscode.CancellationToken): vscode.ProviderResult<vscode.TextEdit[]> {
-    return [];
+    // call it on the entire range
+    let fullRange = new vscode.Range(0, 0, document.lineCount, 0);
+    return this.provideDocumentRangeFormattingEdits(document, fullRange, options, token);
   }
 
+
   provideDocumentRangeFormattingEdits(document: vscode.TextDocument, range: vscode.Range, options: vscode.FormattingOptions,
                                       token: vscode.CancellationToken): vscode.ProviderResult<vscode.TextEdit[]> {
-    return [];
+    // console.log("Formatting");
+    const config = vscode.workspace.getConfiguration('lc2k');
+    const enable = config.get('format.enable');
+    if (!enable) {
+      return [];
+    }
+    const fillOffset = +config.get('format.filloffset');
+    const jalrOffset = +config.get('format.jalroffset');
+    const noopOffset = +config.get('format.noopoffset');
+    const addNewLine = config.get('format.addNewlineAtEOF');
+    // console.log("jalr offset: " + jalrOffset);
+    // console.log("jalr offset: " + fillOffset);
+
+    var edits = [];
```

Comment:
You can use `let` instead of `var`, it has better scoping behavior. It won't make any difference in this particular case, but it's something to keep in mind.